### PR TITLE
feat(chat-e2e): implemented tests for prompt folders namesakes

### PR DESCRIPTION
**Pull request by nartovm**
**Description:**

Implemented new tests for EPMRTC-2969, EPMRTC-2970, EPMRTC-2971, EPMRTC-2972

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+    await dialTest.step(
+      'Try to rename second folder to the same name',
+      async () => {
+        await folderPrompts.openFolderDropdownMenu(
+          ExpectedConstants.newFolderWithIndexTitle(2),
+        );
+        await promptDropdownMenu.selectMenuOption(MenuOptions.rename);
+        await folderPrompts.editFolderName(duplicatedFolderName);
+        await folderPrompts.getEditFolderInputActions().clickTickButton();
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: can use `await folderPrompts.editFolderNameWithTick(duplicatedFolderName);` instead

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+      await expect
+        .soft(
+          folderPrompts.getFolderByName(duplicatedFolderName),
+          ExpectedMessages.folderIsVisible,
+        )
+        .toBeVisible();
+    });
+
+    await dialTest.step(
+      'Drag & drop "New folder 2" to "New folder 1"',
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: please, add expected result into step description

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+      await folderPrompts.expandFolder(
+        ExpectedConstants.newPromptFolderWithIndexTitle(2),
+      );
+    });
+
+    await dialTest.step('Rename folder 3', async () => {
+      await folderPrompts.openFolderDropdownMenu(
+        ExpectedConstants.newFolderWithIndexTitle(3),
+      );
+      folderDropdownMenu.selectMenuOption(MenuOptions.rename);
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: await is missing here

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+      await folderPrompts.editFolderNameWithTick(
+        ExpectedConstants.newFolderWithIndexTitle(2),
+      );
+    });
+
+    await dialTest.step(
+      'Drag renamed folder to the level where namesake exists',
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: please, add expected result into step description

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+          folderPrompts.getFolderByName(
+            ExpectedConstants.newFolderWithIndexTitle(2),
+          ),
+          ExpectedMessages.folderIsVisible,
+        )
+        .toBeVisible();
+    });
+
+    await dialTest.step(
+      'Drag & drop "New folder 2"  located in "New folder 1" to the root',
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: please, add expected result into step description

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts (hunk 1 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: Done

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts (hunk 2 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: Done

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts (hunk 3 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: Done

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts (hunk 4 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: Done

File: apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts (hunk 5 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/promptFolderNamesakes.test.ts]: Done

**Review by irinakartun — DISMISSED** [review]

**Review by irinakartun — APPROVED** [review]

**Pull request merged**